src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -741,7 +847,9 @@ public void Add(ReadOnlySpan<byte> key, long value)
         public void Add(ReadOnlySpan<byte> key, long value, EncodedKey encodedKey)
         {
             AssertValueAndKeySize(key, value);
-
+            // this overload assumes that a previous call to TryGetValue (where you go the encodedKey
+            // already placed us in the right place for the value)
+            Debug.Assert(_internalCursor._stk[_internalCursor._pos].Header->PageFlags == CompactPageFlags.Leaf);
```

Comment:
Let's add the value of `_internalCursor._stk[_internalCursor._pos].Header->PageFlags` to the assertion message

Reply:
done